### PR TITLE
Also detect pre-subject example groups in `RSpec/LeadingSubject`

### PR DESCRIPTION
**Pull request by pirj**
---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [-] Updated documentation.
* [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: lib/rubocop/rspec/language/node_pattern.rb [hunk 1]
```diff
@@ -23,6 +23,9 @@ module NodePattern
 
         def_node_matcher :let?, Helpers::ALL.block_or_block_pass_pattern
 
+        def_node_matcher :include?,
+                         Includes::ALL.send_or_block_or_block_pass_pattern
+
         def_node_matcher :subject?, Subject::ALL.block_pattern
```
Comment (pirj) [inline lib/rubocop/rspec/language/node_pattern.rb:29]: Side note.
Wondering if we should add support for proper detection of:
```ruby
before(:all, &global)
describe(&block)
shared_examples_for('foo', &block)
```

I dislike this syntax, but we've started supporting it step by step.

I'm thinking of a NoBlockPass cop along the way XD

File: spec/rubocop/cop/rspec/leading_subject_spec.rb [hunk 2]
```diff
@@ -160,4 +160,113 @@ def helper_method
...
+        it_behaves_like 'a good citizen', &block
+
+        subject { described_class.new }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Declare `subject` above any other `it_behaves_like` declarations.
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        block = ->{ }
```
Comment (bquorning) [inline spec/rubocop/cop/rspec/leading_subject_spec.rb:265]: Ideally, the `block` would be below the subject too, right? This makes me thing, do we want to allow _anything_ above the `subject`?

Comment (pirj) [reply spec/rubocop/cop/rspec/leading_subject_spec.rb:265]: I knew this question is coming :D
Didn't want to make this example too different from its siblings by making it a `shared_examples_for '...' do |&block|`.

The style guide [is pretty clear about that](https://rspec.rubystyle.guide/#leading-subject):
> When `subject` is used, it should be the first declaration in the example group.

I'd rather not change the behaviour in a radical way here. I can think of e.g. helper methods or custom matchers, some may prefer to have them in a section above subject.

File: lib/rubocop/rspec/language.rb [hunk 3]
```diff
@@ -40,6 +40,10 @@ def send_pattern
           "(send #{RSPEC} #{node_pattern_union} ...)"
         end
 
+        def send_or_block_or_block_pass_pattern
```
Comment (Darhazer) [inline lib/rubocop/rspec/language.rb:43]: I don't like where this is going, but currently, we do not have a better solution

**Pull request merged**